src/core/include/lattice/hal/default/dcrtpoly-impl.h

```diff
@@ -668,6 +668,33 @@ void DCRTPolyImpl<VecType>::SetValuesToZero() {
         m_vectors[i].SetValuesToZero();
 }
 
+template <typename VecType>
+DCRTPolyImpl<VecType> DCRTPolyImpl<VecType>::SetValuesModSwitch(const DCRTPolyImpl& element, NativeInteger modulus) {
+    if (element.GetNumOfElements() != 1) {
+        OPENFHE_THROW(not_implemented_error, "SetValuesModSwitch is implemented only for a DCRTPoly with one tower.");
+    }
+
+    auto Q = element.GetModulus();
+    this->m_params->SetOriginalModulus(modulus);
+
+    for (size_t i = 0; i < m_vectors.size(); i++) {
```

Comment:
change to something like:

auto input{element.GetElementAtIndex(0)};
input.SetFormat(Format::COEFFICIENT);
double Qmod_double{Q.ConvertToDouble() / modulus.ConvertToDouble()};
size_t size{m_vectors.size()};
size_t rdm(m_params->GetRingDimension());

for (size_t i = 0; i < size; ++i) {
    NativeVector tmp(this->GetRingDimension());
    tmp.SetModulus(modulus);
    for (size_t j = 0; j < rdm; ++j)
        tmp[j] = Integer(static_cast<uint64_t>(std::floor(0.5 + input[j].ConvertToDouble() / Qmod_double))).Mod(modulus);
    m_vectors[i].SetValues(std::move(tmp), Format::COEFFICIENT);
    }

Reply:
Done.